source/custom/nixnet_service.custom.cpp

```diff
@@ -257,8 +268,20 @@ ::grpc::Status NiXnetService::WriteState(::grpc::ServerContext* context, const W
         if (state_id != nixnet_grpc::WriteState::WRITE_STATE_STATE_LIN_DIAGNOSTIC_SCHEDULE_CHANGE) {
           return ::grpc::Status(::grpc::INVALID_ARGUMENT, "StateValue for specified StateID is not set");
         }
-        state_value = request->state_value().lin_diagnostic_schedule_change();
-        break;
+        switch (request->state_value().lin_diagnostic_schedule_change().schedule_enum_case()) {
```

Comment:
Pulling this `switch` block in `GetLinDiagnosticScheduleChangeValue` would make it much more readable.

That would've also helped you see that you missed adding `break` statement on line 285 below. :)

Reply:
Done